### DISCUSSION TITLE
patch.rolling updates

### DISCUSSION THREAD

**Discussion by ariellellouch** [DASDAE/dascore#296] [Ideas]
Hey,

I don't like the smoothing function (rolling) changing the size of the output by Nan values. I also saw Derrick suggesting a similar change in projects. 
I suggest either Gaussian or Savitsky-Golay filter and keep the original input size

Ariel

**Comment by d-chambers** [comment]
Hey @ariellellouch,

> I don't like the smoothing function (rolling) changing the size of the output by Nan values

I agree with you, but in designing the rolling functionality we wanted to maintain parity with Pandas and Xarray which do it this way. Rolling can also be more general for any kind of aggregations, and in some cases dealing with edge effects could require a lot of care so simply refusing to perform the aggregation is the most conservative option.  

That being said, I think implementing separate Gaussian or Savitsky-Golay filters in the filter module that behave the way you describe would be a good idea.  [Here](https://github.com/orgs/DASDAE/projects/2/views/1?pane=issue&itemId=40522225) is a link to the card you referenced.

**Comment by d-chambers** [comment]
Closed by #337 and #338.
